docs/trace/customizing-the-sdk/README.md

```diff
@@ -1,10 +1,114 @@
 # Customizing OpenTelemetry .NET SDK
 
-Talks about customizing the SDK.
+**This doc is work-in-progress.**
 
 ## TracerProvider
 
-// TODO: OpenTelemetry Sdk contents about Tracer configuration.
+As shown in the [getting-started](../getting-started/README.md) doc, a valid
+[`TracerProvider`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#tracer-provider)
+must be configured and built to collect traces with OpenTelemetry .NET Sdk.
+`TracerProvider` holds all the configuration for tracing like samplers,
+processors, etc. Naturally, almost all the customizations must be done on the
+`TracerProvider`.
+
+## Building a TracerProvider
+
+Building a `TracerProvider` is done using `TracerProviderBuilder` which must be
+obtained by calling `Sdk.CreateTracerProviderBuilder()`. `TracerProviderBuilder`
+exposes various methods which configures the provider it is going to build. These
+includes methods like `SetSampler`, `AddProcessor` etc, and are explained in
+subsequent sections of this document. Once configuration is done, calling
+`Build()` on the `TracerProviderBuilder` builds the `TracerProvider` instance.
+Once built, changes to its configuration is not allowed, with the exception of
+adding more processors. In most cases, a single `TracerProvider` is created at
+the application startup, and is disposed when application shuts down.
+
+The snippet below shows how to build a basic `TracerProvider`. This will create
+a provider with default configuration, and is not particularly useful. The
+subsequent sections shows how to build a more useful provider.
+
+```csharp
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+using var tracerProvider = Sdk.CreateTracerProviderBuilder().Build();
+```
+
+## TracerProvider configuration
+
+`TracerProvider` holds the tracing configuration, which includes the following:
+
+1. The list of `ActivitySource`s (aka Tracers) from which traces are collected.
+2. The list of instrumentations enabled via
+   [InstrumentationLibrary](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library).
+3. The list of
+   [Processors](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor)
+4. The
+   [Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md)
+   associated with the traces.
+5. The
```

Comment:
It might be helpful to put a diagram (ASCII or mermaid) like this one https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/README.md#programming-model.

Doesn't have to be in this PR though.